core/txpool/legacypool/list.go

```diff
@@ -166,7 +166,6 @@ func (m *sortedMap) Cap(threshold int) types.Transactions {
 		delete(m.items, (*m.index)[size-1])
 	}
 	*m.index = (*m.index)[:threshold]
-	heap.Init(m.index)
```

Comment:
Could you add a comment after `sort.Sort(*m.index)` along the lines of `// the sorted array is a heap, so no need to reheap after deleting tail items`